### PR TITLE
NMC-533 Set new folder name inline: rebased

### PR DESCRIPTION
**Pull request by skjnldsv**
Rebased version of https://github.com/nextcloud/server/pull/28645

Closes https://github.com/nextcloud/server/pull/28645

### CODE REVIEW

**Comment by szaimen** [comment]
@skjnldsv are the styling changes to the template picker really necessary? How does it look like on mobile now?

File: apps/files/src/views/TemplatePicker.vue [hunk 1]
```diff
@@ -284,9 +280,12 @@ export default {
...
 
 	// Make sure we're relative for the loading emptycontent on top
-	::v-deep .modal-container {
-		position: relative;
-		height: fit-content;
+	// use Modal breakpoint + 1px to not target the mobile query
+	@media only screen and (min-width: calc(1024px / 2 + 1px)) {
+		::v-deep .modal-container {
+			position: relative;
+			height: fit-content !important;
```
[the remark this answers is not in the document]
Comment (szaimen) [reply apps/files/src/views/TemplatePicker.vue:287]: Okay, fine by me then

File: apps/files/src/views/TemplatePicker.vue [hunk 2]
```diff
@@ -24,7 +24,7 @@
 	<Modal v-if="opened"
 		:clear-view-delay="-1"
 		class="templates-picker"
-		size="large"
+		size="normal"
```
[the remark this answers is not in the document]
Comment (szaimen) [reply apps/files/src/views/TemplatePicker.vue:27]: Jup

**Review by juliusknorr — APPROVED** [review]

**Review by szaimen — APPROVED** [review]
tested and seems to work well 👍
(didn't review the code though)

**Comment by skjnldsv** [comment]
@danxuliu can yuo help with the acceptance tests please?

File: apps/files/src/views/TemplatePicker.vue [hunk 3]
```diff
@@ -284,9 +280,12 @@ export default {
...
 	// Make sure we're relative for the loading emptycontent on top
-	::v-deep .modal-container {
-		position: relative;
-		height: fit-content;
+	// use Modal breakpoint + 1px to not target the mobile query
+	@media only screen and (min-width: calc(1024px / 2 + 1px)) {
+		::v-deep .modal-container {
+			position: relative;
+			height: fit-content !important;
+		}
```
Comment (szaimen) [inline apps/files/src/views/TemplatePicker.vue:288]: I think we should be able to reverts this change after https://github.com/nextcloud/server/pull/31916
So rebasing and not changing anything here should work, IIRC.

**Comment by danxuliu** [comment]
> @danxuliu can yuo help with the acceptance tests please?

I would love to, but unless I find a time-turner somewhere it is unlikely that I will be able to do it this week, sorry.

**Comment by danxuliu** [comment]
> > @danxuliu can yuo help with the acceptance tests please?
> 
> I would love to, but unless I find a time-turner somewhere it is unlikely that I will be able to do it this week, sorry.

Acceptance tests should be ready now.

File: apps/files/src/views/TemplatePicker.vue (hunk 3 above)
Comment (szaimen) [reply apps/files/src/views/TemplatePicker.vue]: done

**Comment by szaimen** [comment]
@skjnldsv could you please have a look if all changes are still present? I initially messed up the rebase...

**Comment by danxuliu** [comment]
> Looks like one scenario is still failing: image

Although that scenario is failing that one is not the real failure ;-) In errors like those the acceptance tests try the action again, as you can see from the following line to the ones in the screenshot (`Trying again`). As there is not another error immediately after trying again the action succeeded and the test continued.

The real failure is in the last step:
```
And I see that the file list contains a file named "Subsubfolder" # FileListContext::iSeeThatTheFileListContainsAFileNamed()
      Row for file Subsubfolder in file list could not be found after 100 seconds (NoSuchElementException)
```

The test passes locally, so it looks like a hiccup. I would like to understand why it failed to harden the test, but unfortunately I can not see how it could have happened :-(

Independently of all that, if the last commit does not touch JavaScript files please do not amend it to add the compiled files and add an independent commit instead.

**Comment by szaimen** [comment]
> Independently of all that, if the last commit does not touch JavaScript files please do not amend it to add the compiled files and add an independent commit instead.

Sorry for that. Shall I do this here or is it fine for now?

**Comment by szaimen** [comment]
> The test passes locally, so it looks like a hiccup. I would like to understand why it failed to harden the test, but unfortunately I can not see how it could have happened :-(

do we wanna try to restart drone or hope that it was only a hiccup?

**Comment by danxuliu** [comment]
> > Independently of all that, if the last commit does not touch JavaScript files please do not amend it to add the compiled files and add an independent commit instead.
> 
> Sorry for that. Shall I do this now or is it fine for now?

No problem at all :-) I have split the commit in two, it is fine :+1: 

> > The test passes locally, so it looks like a hiccup. I would like to understand why it failed to harden the test, but unfortunately I can not see how it could have happened :-(
>
> do we wanna try to restart drone or hope that it was only a hiccup?

As I have force pushed to split the commit (twice, because the first time I used the wrong commit author :facepalm:) Drone will run again (I have already stopped the first run corresponding to the commit with the wrong author). So let's see ;-)

**Comment by szaimen** [comment]
Great, thanks! :)

**Comment by szaimen** [comment]
It seems to fail again...

**Comment by danxuliu** [comment]
> It seems to fail again...

It is just another hiccup :shrug: And it did not fail in the same step, but on a previous one (looking for _Subfolder_ rather than for _Subsubfolder_). Hopefully I will be able to find out what is going on at some point...

In any case, from the point of view of the acceptance tests, the pull request could be merged despite the failure.

**Comment by szaimen** [comment]
> In any case, from the point of view of the acceptance tests, the pull request could be merged despite the failure.

Okay 👍

I'll leave that to @skjnldsv  :)

**Comment by szaimen** [comment]
is this readdy to merge?

**Pull request closed without merging**